File: olimpiadas II - 20232/Ejercicios/src/Problem272.java
import java.util.Scanner;

public class Problem272 {

	public static String getQuote(boolean start) {
		return start ? "``" : "''";
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		boolean start = true;

		while (sc.hasNextLine()) {
			String line = sc.nextLine();
			int pos = line.indexOf('"');
			StringBuilder result = new StringBuilder();

			while (pos != -1) {
				result.append(line.substring(0, pos));
				result.append(getQuote(start));
				start = !start;
				line = line.substring(pos + 1);
				pos = line.indexOf('"');
			}

			result.append(line);
			System.out.println(result.toString());
		}
		sc.close();
	}

}

File: platforms/UHunt/answers/Problem272.cpp
#include <iostream>
#include <string>

using namespace std;

string getQuote(bool start) {
    return start ? "``" : "''";
}

int main() {
    bool start = true;
    string line;

    while (getline(cin, line)) {
        unsigned long long pos = line.find('"', 0);
        string result;
        while (pos != string::npos) {
            result.append(line.substr(0, pos));
            result.append(getQuote(start));
            start = !start;
            line = line.substr(pos + 1);
            pos = line.find('"');
        }
        result.append(line);
        cout << result << endl;
    }

    return 0;
}
